src/RegisterHandler.php

```diff
@@ -0,0 +1,143 @@
+<?php declare(strict_types=1);
+
+namespace SilverStripe\TOTP;
+
+use OTPHP\TOTP;
+use OTPHP\TOTPInterface;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Environment;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\MFA\Exception\AuthenticationFailedException;
+use SilverStripe\MFA\Method\Handler\RegisterHandlerInterface;
+use SilverStripe\MFA\Service\EncryptionAdapterInterface;
+use SilverStripe\MFA\Store\StoreInterface;
+
+/**
+ * Handles registration requests using a time-based one-time password (TOTP) with the silverstripe/mfa module.
+ */
+class RegisterHandler implements RegisterHandlerInterface
+{
+    use Configurable;
+
+    /**
+     * The link to SilverStripe user help documentation for this authenticator.
+     * @todo add it
+     *
+     * @config
+     * @var string
+     */
+    private static $user_help_link = '';
+
+    public function start(StoreInterface $store): array
+    {
+        $totp = $this->getTotp($store);
```

Comment:
Do we want to always render a new QR code on "start"? In this case re-calling start (ie. when refreshing the page or failing to validate) it will show the same QR code. It's probably all good, just asking the question.

Reply:
I don't know the answer to this question

Reply:
I'll share it around a bit and then raise a follow up if required.